### PR TITLE
Use central entity tables to list remote repos

### PR DESCRIPTION
**Pull request by JAORMX**
# Summary

This is part of generalizing entity registration, as other providers
will need to work with this mechanism. As part of the migration, this
moved the logic to use our new entity model.

Related-To: https://github.com/stacklok/minder/issues/4331
Closes: https://github.com/stacklok/minder/issues/4420

## Change Type

***Mark the type of change your PR introduces:***

- [ ] Bug fix (resolves an issue without affecting existing features)
- [x] Feature (adds new functionality without breaking changes)
- [ ] Breaking change (may impact existing functionalities or require documentation updates)
- [ ] Documentation (updates or additions to documentation)
- [x] Refactoring or test improvements (no bug fixes or new functionality)

# Testing

***Outline how the changes were tested, including steps to reproduce and any relevant configurations. 
Attach screenshots if helpful.***

# Review Checklist:

- [x] Reviewed my own code for quality and clarity.
- [ ] Added comments to complex or tricky code sections.
- [ ] Updated any affected documentation.
- [ ] Included tests that validate the fix or feature.
- [ ] Checked that related changes are merged.

### CODE REVIEW

File: database/query/entities.sql [hunk 1]
```diff
@@ -74,7 +74,9 @@ LIMIT 1;
 
 -- name: GetEntitiesByType :many
 SELECT * FROM entity_instances
-WHERE entity_instances.entity_type = $1 AND entity_instances.project_id = ANY(sqlc.arg(projects)::uuid[]);
+WHERE entity_instances.entity_type = $1
+    AND entity_instances.provider_id = sqlc.arg(provider_id)
+    AND entity_instances.project_id = ANY(sqlc.arg(projects)::uuid[]);
```
Comment (jhrozek) [inline database/query/entities.sql:79]: 🚀

File: internal/providers/manager/manager.go [hunk 2]
```diff
@@ -33,6 +33,12 @@ import (
 
 //go:generate go run go.uber.org/mock/mockgen -package mock_$GOPACKAGE -destination=./mock/$GOFILE -source=./$GOFILE
 
+// NameProviderTuple is a tuple of a provider name and the provider instance
+type NameProviderTuple struct {
+	Name     string
+	Provider v1.Provider
+}
```
Comment (jhrozek) [inline internal/providers/manager/manager.go:40]: This is a clear improvement. I was actually wondering about changing the generic provider interface to always have to implement `GetName` and `GetID`. But that's neither here or now.

File: internal/repositories/service.go [hunk 3]
```diff
@@ -228,18 +228,28 @@ func (r *repositoryService) CreateRepository(
...
+		ProviderID: providerID,
+		Projects:   []uuid.UUID{projectID},
+	})
+	if err != nil {
+		return nil, fmt.Errorf("error fetching repositories: %w", err)
+	}
+
+	ents := make([]*models.EntityWithProperties, 0, len(repoEnts))
+	for _, ent := range repoEnts {
+		ewp, err := r.propSvc.EntityWithProperties(ctx, ent.ID, nil)
```
Comment (jhrozek) [inline internal/repositories/service.go:244]: should this be a transaction so that handlers are not interrupted in case the repo goes away?

**Review by jhrozek — APPROVED** [review]
approved with two suggestions

File: internal/providers/manager/manager.go (hunk 2 above)
Comment (JAORMX) [reply internal/providers/manager/manager.go:40]: That would be really handy, let's do it in a subsequent PR.

**Review by rdimitrov — APPROVED** [review]

**Pull request merged**